src/utilities/notifications.ts

```diff
@@ -20,3 +43,75 @@ export function getPostUpdatedNotificationContent( userState: UserState, status:
 
 	return `${ noun } ${ verb } by ${ userState.name }.`;
 }
+
+/**
+ * Get the content of a user presence notification, based on the type.
+ *
+ * @param userState the user state of the user related to the notification
+ * @param type the type of notification
+ * @returns the content of the user presence notification
+ */
+export function getUserPresenceNotificationContent(
+	userState: UserState,
+	type: NotificationType
+): string {
+	let action = type === NotificationType.UserEntered ? 'entered' : 'exited';
+	return `${ userState.name } has ${ action } the post.`;
+}
+
+function shouldSendNotification( userState: UserState, type: NotificationType ): boolean {
+	// If notifications for user joining is disabled, skip.
+	if (
+		type === NotificationType.UserEntered &&
+		! select( settingsStore ).getSetting( Setting.USER_ENTER_NOTIFICATION )
+	) {
+		return false;
+	}
+
+	// If notifications for user leaving is disabled, skip.
+	if (
+		type === NotificationType.UserExited &&
+		! select( settingsStore ).getSetting( Setting.USER_EXIT_NOTIFICATION )
+	) {
+		return false;
+	}
+
+	// If the current user is the one who joined/left, skip.
+	if (
+		( type === NotificationType.UserEntered || type === NotificationType.UserExited ) &&
+		userState.isMe
+	) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Send a notification to the editor.
+ *
+ * Certain notifications can be skipped based on user settings, or scenarios.
+ *
+ * @param content the notification content
+ * @param userState the user state of the user related to the notification
+ * @param type the type of notification
+ */
+export function sendNotification(
```

Comment:
I didn't think it's critical if these notifications are sent or not, in the event that they are skipped. Hence why its void - treat it like a send it and forget it type of function. I don't think this needs to be a boolean right now, we can always revisit this later.